lib/flowViewport/sceneIndex/fvpSelectionSceneIndex.cpp

```diff
@@ -132,7 +132,12 @@ SelectionSceneIndex::GetPrim(const SdfPath &primPath) const
         .Msg("SelectionSceneIndex::GetPrim() called.\n");
 
     HdSceneIndexPrim result = GetInputSceneIndex()->GetPrim(primPath);
-    if (!result.dataSource) {
+    // An empty data source can be taken to mean an invalid prim, but it can
+    // also be an ancestor in a path to an actual prim (see
+    // https://forum.aousd.org/t/representation-of-invalid-hdsceneindexprim/833/2
+    // ).  Such an ancestor prim can be selected, even if typeless, so add
+    // a selection data source in such a case.
+    if (!result.dataSource && !_selection->IsFullySelected(primPath)) {

```

Comment:
The bug fix.  The crash was caused by a failing assertion in the wireframe selection highlighting scene index, triggered by the fact that switching selection to a prim without a data source did not add a selection data source to that prim.